### PR TITLE
add a BillingClient

### LINKED ISSUE #32: Need to add a BillingClient with associated options and responses to DeepgramClient
The implementation for the `/billing` endpoint should mirror the design of other endpoints, like the `IKeyClient`, `KeyClient`, and associated request and response options.

The new `BillingClient` should contain the following methods:

- [ ] `GetAllBalancesAsync(string projectId)`
- [ ] `GetBalanceAsync(string projectId, string balanceId)`

Documentation for the `/billing` endpoint can be found at https://developers.deepgram.com/api-reference/#billing

### PR DESCRIPTION
**Pull request by NemrudDemir**
resolves #32

### CODE REVIEW

File: Deepgram/Billing/BillingClient.cs [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        {
+            _credentials = credentials;
+        }
+
+        /// <summary>
+        /// Generates a list of outstanding balances for the specified project. To see balances, the authenticated account must be a project owner or administrator
+        /// </summary>
+        /// <param name="projectId">Unique identifier of the project for which you want to retrieve outstanding balances</param>
+        /// <returns>List of Deepgram balances</returns>
+        public async Task<Billing[]> GetAllBalancesAsync(string projectId)
```
Comment (MichaelJolley) [inline Deepgram/Billing/BillingClient.cs:21]: This return type is incorrect. If you look at the documentation, the API doesn't return an array of `Billing` objects. It instead returns an object with a `balances` property that is an array of `Billing` objects.

```js
{
  "balances": [
    {
      "balance_id": "uuid",
      "amount": 0,
      "units": "string",
      "purchase": "string"
    }
  ]
}
```

I believe we've handled things like that in the past with a "List" object. (e.g. BillingList class with a singular property called Balances that is the array of Billing objects.)

File: Deepgram/Billing/IBillingClient.cs [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+namespace Deepgram.Billing
+{
+    public interface IBillingClient
+    {
+        /// <summary>
+        /// Generates a list of outstanding balances for the specified project. To see balances, the authenticated account must be a project owner or administrator
+        /// </summary>
+        /// <param name="projectId">Unique identifier of the project for which you want to retrieve outstanding balances</param>
+        /// <returns>List of Deepgram balances</returns>
+        Task<Billing[]> GetAllBalancesAsync(string projectId);
```
Comment (MichaelJolley) [inline Deepgram/Billing/IBillingClient.cs:12]: Will need to update based on the BillingClient feedback.

**Review by MichaelJolley — CHANGES_REQUESTED** [review]
Good start @NemrudDemir. Couple changes needed before we can merge.

File: Deepgram/Billing/IBillingClient.cs (hunk 2 above)
Comment (NemrudDemir) [reply Deepgram/Billing/IBillingClient.cs]: Thank you for your detailed comment. I've added a BillingList-Object as you mentioned.

**Review by MichaelJolley — APPROVED** [review]
Looks good @NemrudDemir. Thanks for the update!

**Pull request merged**